quaint/src/connector/sqlite/error.rs

```diff
@@ -199,6 +199,32 @@ impl From<rusqlite::Error> for Error {
                     builder.build()
                 }
             },
+
+            rusqlite::Error::SqlInputError {
+                error: ffi::Error { extended_code, .. },
+                ref msg,
+                ..
+            } => match msg {
+                d if d.starts_with("no such column: ") => {
+                    let column = d.split("no such column: ").last().into();
+                    let kind = ErrorKind::ColumnNotFound { column };
+
+                    let mut builder = Error::builder(kind);
+                    builder.set_original_code(format!("{extended_code}"));
+                    builder.set_original_message(d);
+
+                    builder.build()
+                }
+                _ => {
+                    let description = msg.clone();
+                    let mut builder = Error::builder(ErrorKind::QueryError(e.into()));
+                    builder.set_original_code(format!("{extended_code}"));
+                    builder.set_original_message(description);
+
+                    builder.build()
+                }
+            },
+
```

Comment:
Why is this change needed?

Reply:
Underlying SQLite changes, they've changed how their messages are being emitted.

Reply:
Just to clarify, there are still some match branches above this under the previous error type that may or may not but necessary anymore but we thought best to leave them in just in case

Reply:
@janpio 

> Can you quickly ELI5 the specific change they introduce in which case?

SQLite 3.38 introduced new errors that reference a specific token in the input SQL. In `rusqlite` they are represented as a new error type, `SqlInputError` (as opposed to the old type, `SqliteFailure`, which doesn't contain this information and only has the error code and message).

I suspect that other cases that weren't copied to the `match` arm for the new type are probably relevant here as well. We should add tests for those and check.

Reply:
"no such table" is covered by `tests::query::error::table_does_not_exist`, and "has no column named" is covered by `tests::query::error::column_does_not_exist_on_write`. So currently this PR handles errors correctly, and we have the tests that should catch if that changes in a future rusqlite update.